src/metabase/upload/types.cljc

```diff
@@ -0,0 +1,305 @@
+(ns metabase.upload.types
+  (:refer-clojure :exclude [make-hierarchy])
+  (:require
+   [clojure.string :as str]
+   [metabase.upload.parsing :as upload-parsing]
+   [metabase.util :as u]
+   [metabase.util.malli :as mu]
+   [metabase.util.ordered-hierarchy :as ordered-hierarchy :refer [make-hierarchy]]))
+
+;;;; +------------------+
+;;;; | Schema detection |
+;;;; +------------------+
```

Comment:
I think this block is unnecessary now that there's a whole namespace for schema detection